[KennyLewi] iP

KennyLewi opened the pull request:
# Peridot Bot
>Lets go peridot
:shipit: 

**Peridot can:**
- Save your tasks
- List out your tasks

**And you can categorize your tasks into:**
1.Events
2. Todo(s)
3. Deadlines

For example, you can use the CLI to input tasks
```java
event Exams /from 2024-10-12 /to 2024-10-13
```
Here `event` specifies that you want to add an 'Event' type task to the list

You can also:
- [x] mark tasks
- [ ] unmark tasks

> [!CAUTION]
> This bot is very ~~bad~~ good

[Thanks for reading](https://i.scdn.co/image/ab67616d0000b273d9cd13b2d78be62fc76cb111)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

KennyLewi committed:
renamed chatbot to peridot, and added greeting

KennyLewi committed:
Added echo response, and bot name before each bot message

KennyLewi committed:
added a list (you can add tasks and display all stored tasks)

KennyLewi committed:
added marking of tasks, and used a new class 'Task'

KennyLewi committed:
added Todo, Events, and Deadlines, and moved the switch command to List.java

KennyLewi committed:
added testing

KennyLewi committed:
added error handling

KennyLewi committed:
used arraylist to store data instead of Task array

KennyLewi committed:
added task deletion

KennyLewi committed:
Add ability to read locally saved data

KennyLewi committed:
Add writing to save new tasks

KennyLewi committed:
Merge branch 'branch-Level-7'

KennyLewi committed:
Change time format from String to Java LocalDate

KennyLewi committed:
Merge branch 'branch-Level-8'

KennyLewi committed:
Make the code more OOP

KennyLewi committed:
Update gitignore to ignore TaskList data

KennyLewi committed:
Organize into packages

KennyLewi committed:
Merge branch 'add-gradle-support'

KennyLewi committed:
Use Gradle

KennyLewi committed:
Add JUnit tests

KennyLewi committed:
Create a JAR File

KennyLewi committed:
Add JavaDoc Comments

KennyLewi committed:
Follow the Coding Standard

KennyLewi committed:
Add find method to tasklist

KennyLewi committed:
Merge branch 'branch-A-JavaDoc'

KennyLewi committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Parser.java

KennyLewi committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/chatbot/Parser.java
#	src/main/java/chatbot/Ui.java
#	src/main/java/storage/Storage.java
#	src/main/java/tasks/Task.java
#	src/test/java/chatbot/ParserTest.java

Zenkoh1 commented on src/main/java/chatbot/Ui.java:
```diff
@@ -0,0 +1,48 @@
+        botSay("Bye!");
+    }
+
+    /**
+     * Prints out the message with a "Peridot: " prefix.
+     */
```
Not adhering to JavaDoc specifications (no @param etc.).

Zenkoh1 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,58 @@
+
+public class Task {
+
+    private boolean done;
```
```suggestion
    private boolean isDone;
```

Zenkoh1 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+
+public class TaskList {
+
+    private List<Task> list;
```
You can be more descriptive about what `list` means.

Zenkoh1 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     * @param task Task to be added.
+     * @param saved True if the task to added should be pre-marked as done, otherwise False.
+     */
+    public void add(Task task, boolean saved) {
```
```suggestion
    public void addTask(Task task, boolean saved) {
```

Same as the above, can be more descriptive with your method names.

KennyLewi committed:
Use Checkstyle

KennyLewi committed:
Merge branch 'branch-A-CheckStyle'

KennyLewi committed:
Add simple GUI that replies with 'hello'

KennyLewi committed:
Fix GUI to properly respond to user input

KennyLewi committed:
Merge branch 'branch-Level-10'

Plishh commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
```
I like how all imports explicitly listed 👍

Plishh commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+                            LocalDate.parse(start), LocalDate.parse(end), done));
+                    break;
+                default:
+                    break;
```
Does default case need a break?

Plishh commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+            return loadedData;
+        } catch (FileNotFoundException e) {
+            System.out.println("Scanner failed in load() method of storage.Storage-type object");
+            return null;
```
Do you need to return anything after catching an exception?

Plishh commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,58 @@
+
+    public boolean isDone() {
+        return this.done;
+    }
```
Maybe name its getIsDone()?

Plishh reviewed (commented):
THe code overall looks good to me 👍

KennyLewi committed:
Add assertions to check developer errors

There are several possible developer erros such as not initialising task list,
not updating the task list correctly. Assertions can be used to make sure that
these errors can be caught

KennyLewi committed:
Add more javadoc comments to improve documentation

KennyLewi committed:
Add more javadoc comments to improve documentation

KennyLewi committed:
Shorten method length 'answer' in Parser.java

Improved readability by moving out the try-catch block in the method 'answer'
in Parser.java so that there is no need repeat the try-catch block multiple times

KennyLewi committed:
Merge branch 'branch-A-Assertions'

KennyLewi committed:
Revert "Merge branch 'branch-A-Assertions'"

This reverts commit f065319d9669dd993d68b3315b70bfe2cd0da6f5, reversing
changes made to 6b0dcc9dc71323205db771bc6e776cafa60fa226.

KennyLewi committed:
Merge pull request #1 from KennyLewi/branch-A-CodeQuality

Add assertions to the iP

KennyLewi committed:
Merge branch 'master' into branch-A-Assertions

KennyLewi committed:
Merge pull request #2 from KennyLewi/branch-A-Assertions

Merge branch 'master' into branch-A-Assertions

KennyLewi committed:
Add help command

KennyLewi committed:
Merge pull request #3 from KennyLewi/branch-BCD-Extension

Add help command

KennyLewi committed:
Add JavaDoc comments to improve documentation.

Add JavaDoc comments to the class Peridot.java to improve the documentation,
and improve ease of usage of code

KennyLewi committed:
Add JavaDoc comments to Parser class

There are methods in the Parser class that is not documented with
JavaDoc comments.

Adding JavaDoc comments would improve the documentation.

As a step towards better documentation, let's add JavaDoc comments to
the Parser class. Doing so will make subsequent function calls easier
as there are JavaDoc comments to refer to.

KennyLewi committed:
Add JavaDoc comments to TaskList Class

There are methods in the TaskList class that is not documented with
JavaDoc comments.

Adding JavaDoc comments would improve the documentation.

As a step towards better documentation, let's add JavaDoc comments to
the TaskList class. Doing so will make subsequent function calls easier
as there are JavaDoc comments to refer to.

KennyLewi committed:
Add JavaDoc comments to Storage class

There are methods in the Storage class that is not documented with
JavaDoc comments.

Adding JavaDoc comments would improve the documentation.

As a step towards better documentation, let's add JavaDoc comments to
the Storage class. Doing so will make subsequent function calls easier
as there are JavaDoc comments to refer to.

KennyLewi committed:
Add JavaDoc comments to Deadline class

There are methods in the Deadline class that is not documented with
JavaDoc comments.

Adding JavaDoc comments would improve the documentation.

As a step towards better documentation, let's add JavaDoc comments to
the Deadline class. Doing so will make subsequent function calls easier
as there are JavaDoc comments to refer to.

KennyLewi committed:
Add JavaDoc comments to Todo class

There are methods in the Todo class that is not documented with
JavaDoc comments.

Adding JavaDoc comments would improve the documentation.

As a step towards better documentation, let's add JavaDoc comments to
the Todo class. Doing so will make subsequent function calls easier
as there are JavaDoc comments to refer to.

KennyLewi committed:
Update README.md

README.md is outdated with old Duke information.

Updating the README will improve usability.

As a step towards better usability, let's update the README.md.

KennyLewi committed:
Add more exceptions

Some possible user errors are not handled.

Adding more exceptions to handle these will improve error detection.

As a step towards better usability, let's add exceptions.

KennyLewi committed:
Add JavaDoc comments to multiple classes

There are methods that are not documented with JavaDoc comments.

Adding JavaDoc comments would improve the documentation.

As a step towards better documentation, let's add JavaDoc comments.
Doing so will make subsequent function calls easier
as there are JavaDoc comments to refer to.

KennyLewi committed:
Add JavaDoc comments to methods in Parser class

There are methods in the Parser class that is not documented with
JavaDoc comments.

Adding JavaDoc comments would improve the documentation.

As a step towards better documentation, let's add JavaDoc comments to
the Parser class. Doing so will make subsequent function calls easier
as there are JavaDoc comments to refer to.

KennyLewi committed:
Add more exceptions

Some possible user errors are not handled.

Adding more exceptions to handle these will improve error detection.

As a step towards better usability, let's add exceptions.
